test/snapshots/prefer-dom-node-dataset.mjs.md

```diff
@@ -269,6 +269,22 @@ Generated by [AVA](https://avajs.dev).
         | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer \`.dataset\` over \`setAttribute(…)\`.␊
     `
 
+## Invalid #17
+      1 | element?.setAttribute("data-optional-chaining", "foo");
+
+> Output
+
+    `␊
+      1 | element.dataset.optionalChaining = "foo";␊
```

Comment:
We can't fix to this, it will break code.

Reply:
Hello Fisk, thanks!
How exactly can I help?

Reply:
We should either remove the fix for optional members or add a safe fix.

Reply:
I'm sorry @fisker, it took me a while to understand what was wrong, I think I got it now:

We should expect the autofix feature to keep the optional chaining, is that it?

Like 

 ```suggestion
       1 | element?.dataset.optionalChaining = "foo";␊
```
Did I get it right?

Thanks again for your review, cheers

Reply:
I don't think so, optional chaining can't be LHS.

Reply:
1. `element?.setAttribute('data-foo', 'bar')`

Since `foo?.dataset.foo = 'bar'` is invalid, and we don't want auto fix to wrap it in a `if`, so we should just remove the autofix.

2. `element?.getAttribute('data-foo')`

Should be safe to fix to `element?.dataset.foo`

3. `element?.removeAttribute('data-foo')`

Should be safe to fix to `delete element?.dataset.foo`

4. `element?.hasAttribute('data-foo')`

Since `Object.hasOwn(element?.dataset, 'foo')` can cause runtime error, so I guess we should remove the autofix too.

Reply:
I'm adding `hasOptionalChainElement` utility in #2169, you can use it after we merge.